diff -u coq-lib-orig/lem_list.v coq-lib/lem_list.v
--- coq-lib-orig/lem_list.v	2014-02-13 15:15:21.000000000 +0000
+++ coq-lib/lem_list.v	2014-02-13 15:22:21.000000000 +0000
@@ -158,7 +158,7 @@
 end. *)
 (* [?]: removed value specification. *)
  (* before also called "flatten" *)
-Definition concat {a : Type}   : list (list a) -> list a:=  List.fold_right List.append [].
+Definition concat {a : Type}   : list (list a) -> list a:=  List.fold_right (@List.app _) []. (* XXX See issue #97. *)
 (* [?]: removed value specification. *)
 
 (* [?]: removed top-level value definition. *)
@@ -218,7 +218,7 @@
 Program Fixpoint genlist {a : Type}  (f : nat  -> a) (n : nat )  : list a:= 
   match ( n) with 
     | 0 => []
-    | S (n') =>  List.append(genlist f n') [(f n')]
+    | S (n') =>  (@List.app _)(genlist f n') [(f n')] (* XXX See issue #97. *)
   end.
 (* [?]: removed value specification. *)
 
diff -u coq-lib-orig/lem_num.v coq-lib/lem_num.v
--- coq-lib-orig/lem_num.v	2014-02-13 15:15:21.000000000 +0000
+++ coq-lib/lem_num.v	2014-02-13 15:22:07.000000000 +0000
@@ -273,6 +273,7 @@
 
 (* [?]: removed value specification. *)
 
+(* XXX "Cannot guess decreasing argument of fix" (see issue #110).
 Program Fixpoint gen_pow_aux {a : Type}   (mul : a ->  a ->  a)  (a0 : a)  (b : a)  (e : nat )  : a:= 
    match ( e) with 
      | 0 => a0 (* cannot happen, call discipline guarentees e >= 1 *)
@@ -286,6 +287,7 @@
   if nat_ltb e(id 0) then one else 
   if ( classical_boolean_equivalence e(id 0)) then one else gen_pow_aux mul one b e.
 (* [?]: removed value specification. *)
+*)
 
 
 Instance x90_NumPow : NumPow nat := { 
@@ -472,9 +474,11 @@
 (* [?]: removed value specification. *)
  (* TODO: check *)
 
+(*
 Instance x69_NumAbs : NumAbs Z := {
    abs  :=  Zabs_nat
 }.
+*)
 
 (* [?]: removed value specification. *)
 
@@ -983,7 +987,7 @@
 Definition int32FromInt  (i : Z ) :=  int32FromInteger (id i).
 (* [?]: removed value specification. *)
 
-Definition int32FromInt64  (i : Z ) :=  int32FromInteger (TODO i).
+(* Definition int32FromInt64  (i : Z ) :=  int32FromInteger (TODO i). *) (* XXX See issue #116. *)
 (* [?]: removed value specification. *)
 
 (* [?]: removed value specification. *)
@@ -999,7 +1003,7 @@
 Definition int64FromInt  (i : Z ) :=  int64FromInteger (id i).
 (* [?]: removed value specification. *)
 
-Definition int64FromInt32  (i : Z ) :=  int64FromInteger (TODO i).
+(* Definition int64FromInt32  (i : Z ) :=  int64FromInteger (TODO i). *) (* XXX See issue #116. *)
 (* [?]: removed value specification. *)
 
 (* [?]: removed value specification. *)
diff -u coq-lib-orig/lem_word.v coq-lib/lem_word.v
--- coq-lib-orig/lem_word.v	2014-02-13 15:15:21.000000000 +0000
+++ coq-lib/lem_word.v	2014-02-13 15:22:07.000000000 +0000
@@ -3,7 +3,7 @@
 Require Import Arith.
 Require Import Bool.
 Require Import List.
-Require Import String.
+(* Require Import String. *) (* XXX See issue #97. *)
 Require Import Program.Wf.
 
 Open Scope nat_scope.
@@ -35,7 +35,7 @@
    option  nat    ->  (* length of the sequence, Nothing means infinite length *)
    bool  ->        (* sign of the word, used to fill up after concrete value is exhausted *)
    list  bool  -> bitSequence .
-Definition bitSequence_default: bitSequence  := BitSeq maybe_default bool_default list_default.
+Definition bitSequence_default: bitSequence  := BitSeq option_default bool_default list_default.
 (* [?]: removed value specification. *)
 
 
@@ -110,11 +110,12 @@
 
 (* [?]: removed value specification. *)
 
-Program Fixpoint bitSeqBinopAux  (binop : bool  -> bool  -> bool ) (s1 : bool ) (bl1 : list (bool )) (s2 : bool ) (bl2 : list (bool )) := 
+(* XXX See issue #117. *) 
+Fixpoint bitSeqBinopAux  (binop : bool  -> bool  -> bool ) (s1 : bool ) (bl1 : list (bool )) (s2 : bool ) := fix bitSeqBinopAux' (bl2 : list (bool )) := 
   match ( (bl1,  bl2)) with 
     | ([],  []) => []
     | (b1 :: bl1',  []) => (binop b1 s2) :: bitSeqBinopAux binop s1 bl1' s2 []
-    | ([],  b2 :: bl2') => (binop s1 b2) :: bitSeqBinopAux binop s1 []   s2 bl2'
+    | ([],  b2 :: bl2') => (binop s1 b2) :: bitSeqBinopAux' bl2'
     | (b1 :: bl1',  b2 :: bl2') => (binop b1 b2) :: bitSeqBinopAux binop s1 bl1' s2 bl2'
   end.
 
@@ -173,6 +174,7 @@
 (* [?]: removed value specification. *)
 
 
+(* XXX Can't determine termination (see issue #117).
 Program Fixpoint boolListFromNatural  (acc : list (bool ))  (remainder : nat ) := 
  if ( nat_gtb remainder(id 0)) then 
    (boolListFromNatural (( classical_boolean_equivalence( nat_mod remainder(id 2))(id 1)) :: acc) 
@@ -190,6 +192,7 @@
 Definition bitSeqFromInteger  (len_opt : option (nat ) ) (i : Z ) :=  
   match ( boolListFromInteger i) with (s,  bl) =>
     resizeBitSeq len_opt (BitSeq None s bl) end.
+*)
 (* [?]: removed value specification. *)
 
 Definition integerFromBitSeq  (bs : bitSequence ) :=  
@@ -197,6 +200,7 @@
     integerFromBoolList (s,  bl) end.
 (* [?]: removed value specification. *)
 
+(* XXX Uses commented-out stuff from above.
 Definition bitSeqArithUnaryOp  (uop : Z  -> Z ) (bs : bitSequence ) :=  
   match ( bs) with ( BitSeq len _ _) =>
     bitSeqFromInteger len (uop (integerFromBitSeq bs)) end.
@@ -212,6 +216,7 @@
 (* [?]: removed value specification. *)
 
 Definition bitSeqArithBinTest {a : Type}  (binop : Z  -> Z  -> a) (bs1 : bitSequence ) (bs2 : bitSequence )  : a:=  binop (integerFromBitSeq bs1) (integerFromBitSeq bs2).
+*)
 (* [?]: removed value specification. *)
 
 (* [?]: removed top-level value definition. *)
@@ -231,6 +236,7 @@
 
 (* [?]: removed value specification. *)
 
+(*
 Definition bitSeqLess  (bs1 : bitSequence ) (bs2 : bitSequence ) :=  bitSeqArithBinTest int_ltb bs1 bs2.
 (* [?]: removed value specification. *)
 
@@ -759,3 +765,4 @@
 }.
 
 
+*)
\ No newline at end of file
